pkg/process/checks/net.go

```diff
@@ -227,10 +225,10 @@ func (c *ConnectionsCheck) shouldRunCheck(start time.Time) bool {
 	}
 
 	// Dynamic interval mode: use capacity-aware logic
-	timeSinceLastRun := start.Sub(c.lastFullRunTime)
+	timeSinceLastRun := start.Sub(c.lastRunTime)
 	// Add a small tolerance (250ms) to account for differences in the check interval and the guaranteed run interval
-	guaranteedIntervalWithTolerance := c.guaranteedRunInterval - (time.Millisecond * 250)
-	isTimeForGuaranteedRun := c.lastFullRunTime.IsZero() || timeSinceLastRun >= guaranteedIntervalWithTolerance
+	guaranteedIntervalWithTolerance := ConnectionsCheckDefaultInterval - (time.Millisecond * 250)
```

Comment:
Will do the math correctly? i.e., would `ConnectionsCheckDefaultInterval.Sub(250 * time.Millisecond)` be more correct here?

Reply:
I will update it, it worked in the e2e tests though